lib/ansible/modules/cloud/amazon/cloudformation_facts.py

```diff
@@ -256,15 +262,21 @@ def main():
     service_mgr = CloudFormationServiceManager(module)
     stack_name = module.params.get('stack_name')
     result = {
-        'ansible_facts': { 'cloudformation': { stack_name:{} } }
+        'ansible_facts': {'cloudformation': {stack_name: {}}}
     }
     facts = result['ansible_facts']['cloudformation'][stack_name]
     facts['stack_description'] = service_mgr.describe_stack(stack_name)
 
+    if not facts['stack_description']:
+        # stack does not exist; return expected (empty) output + message
+        result.update(msg="Stack %s does not exist" % stack_name)
+        facts.pop('stack_description')
+        result['changed'] = False
+        module.exit_json(**result)
+
     # Create stack output and stack parameter dictionaries
-    if facts['stack_description']:
-        facts['stack_outputs'] = to_dict(facts['stack_description'].get('Outputs'), 'OutputKey', 'OutputValue')
-        facts['stack_parameters'] = to_dict(facts['stack_description'].get('Parameters'), 'ParameterKey', 'ParameterValue')
+    facts['stack_outputs'] = to_dict(facts['stack_description'].get('Outputs'), 'OutputKey', 'OutputValue')
```

Comment:
I'm thinking these parameters shouldn't be filled in, since we're popping `stack_description` so if a user wanted a hard-fail for nonexistent stacks, they could do:

```
- cloudformation_facts:
    stack_name: not-real
  failed_when: not cloudformation['not-real']
```